### PR TITLE
Fix constraint file creation and clean up logic on Windows

### LINKED ISSUE #63: Plugin install on Windows a `WinError 32` show up related to some constraints file when exiting napari
Trigger an operation from the plugin dialog and, at napari exit, some error related to not having access to a constraint file appears:

# A preview

winerror_plugin_dialog

# Traceback/error message

```
Exception ignored in atexit callback: <built-in function unlink>
PermissionError: [WinError 32] El proceso no tiene acceso al archivo porque está siendo utilizado por otro proceso: 'C:\\Users\\dalth\\AppData\\Local\\Temp\\tmpx5py0rxw-napari-constraints.txt'
```

### PR DESCRIPTION
**Pull request by dalthviz**
Fix constraing file creation and clean up logic by using `tempfile.mkstemp` returned file descriptor to handle constraint file operations. The previous logic was ignoring the returned file descriptor and instead directly using `open` via the temporary file path and missing to close the ignored file descriptor

Fixes #63 

Edit: Move from using `tempfile.mkstemp` to `tempfile.NamedTemporaryFile`

### CODE REVIEW

File: napari_plugin_manager/qt_package_installer.py [hunk 1]
```diff
@@ -140,8 +140,8 @@ def environment(
     @classmethod
     @lru_cache(maxsize=0)
     def _constraints_file(cls) -> str:
-        _, path = mkstemp("-napari-constraints.txt", text=True)
-        with open(path, "w") as f:
+        file_descriptor, path = mkstemp("-napari-constraints.txt", text=True)
+        with os.fdopen(file_descriptor, "w") as f:
```
Comment (jaimergp) [inline napari_plugin_manager/qt_package_installer.py:144]:
```suggestion
        with NamedTemporaryFile("-napari-constraints.txt", delete=False) as f:
```

File: napari_plugin_manager/qt_package_installer.py [hunk 2]
```diff
@@ -140,8 +140,8 @@ def environment(
...
             f.write("\n".join(cls.constraints()))
         atexit.register(os.unlink, path)
```
Comment (jaimergp) [inline napari_plugin_manager/qt_package_installer.py:146]:
```suggestion
        atexit.register(os.unlink, f.name)
```

**Review by jaimergp — APPROVED** [review]

**Review by goanpeca — APPROVED** [review]

**Comment by dalthviz** [comment]
Should I merge this @jaimergp @goanpeca ? Or maybe although it got approved something is missing?

**Comment by jaimergp** [comment]
Go for it!

**Pull request merged**